lib/Mojo/Promise.pm

```diff
@@ -56,7 +56,20 @@ sub catch { shift->then(undef, shift) }
 
 sub clone { $_[0]->new->ioloop($_[0]->ioloop) }
 
-sub finally { shift->_finally(1, @_) }
+sub finally {
+  my $class = ref $_[0];
+  my $cb    = $_[1];
+
+  my $big_cb = sub {
+    my @vals = @_;
+
+    $class->resolve($cb->())->then(sub {@vals});
```

Comment:
Pretty sure `$class->resolve` does not consider the current `ioloop` being used.

Reply:
Are you able to suggest a correction here? I don’t know what else would be the proper syntax; I’m looking for the equivalent of `Promise.resolve(…)`, which I thought was Mojo::Promise->resolve().

Reply:
`Mojo::Promise` instances unfortunately carry an `ioloop` property around, that needs to be inherited by all chained promises.

Reply:
Very unlike JavaScript promises, in Mojolicious applications there can be multiple event loops active at the same time.

Reply:
And note that i don't believe that's a good idea, in fact i voted against it... but now it just is what it is and we have to deal with it.